### PR TITLE
FINERACT-1194: Move Gradle wrapper to root folder

### PR DESCRIPTION
**Pull request by vidakovic**
## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by vidakovic** [comment]
@vorburger So... these are the bare minimum changes to make Gradle work from the root folder. All other changes were just follow-ups due to errors on the console.

**Comment by vidakovic** [comment]
@vorburger @ptuomola Do you have an idea why Docker is not finding the JAR file? I've checked and it is built actually... but I can't find how the JAR file is added to the Docker image... any ideas appreciated.

**Comment by ptuomola** [comment]
Hi @vidakovic - looks like after your changes, the Dockerfile builds the fineract-provider.jar into /fineract/fineract/build/libs, but tries to run it from /fineract/build/libs/fineract-provider*.jar. So that doesn't work..

**Comment by vidakovic** [comment]
@ptuomola Could you lend me a helping hand here to get this PR over the finish line? Have to admit that I'm having a bit of a hard time navigating the build process here with copying everything into the image and running the Gradle build again; not really sure which moving part creates that additional folder. Thanks a ton.

**Comment by ptuomola** [comment]
Hi @vidakovic - I don't think this is anything specific to Docker, I think the same problem exists even if you run the build outside docker:

Let's say I've cloned the repository to /Users/ptuomola/Development/apache-fineract/fineract

When I run ./gradlew -PautomatedBuild=true --no-daemon -q -x rat -x test bootJar in that directory, the JAR file gets created to:

/Users/ptuomola/Development/apache-fineract/fineract/Users/ptuomola/Development/apache-fineract/fineract/build/libs

Looking at your changes in build.gradle, I think $rootDir is being provided as an absolute path but being treated by gradle as a relative path... 

But I suppose the question is do we even need $rootDir in those places - looks like the path would be correct if you left it out completely?

**Comment by vorburger** [comment]
@vidakovic I need to fix FINERACT-1195 before I can help you fix the broken Docker build here.. will do ASAP.

**Comment by ptuomola** [comment]
As per my above comment, it should be as simple as removing $rootDir references from build.gradle... but I wanted @vidakovic to confirm that he thinks that makes sense, as I'm not sure what else may be impacted.

**Comment by vidakovic** [comment]
@ptuomola Good idea with rootdir... will try that. Thanks again

**Comment by ptuomola** [comment]
Great stuff... and just to be clear I don't think you need to remove all references to $rootDir, maybe it's enough just to remove the one used for setting buildDir?

File: fineract-provider/build.gradle
Comment (ptuomola) [inline fineract-provider/build.gradle]:
```suggestion
buildDir = new File(rootProject.projectDir, "build")
```

Comment (ptuomola) [reply fineract-provider/build.gradle]: Just tried this change and it seems to fix both Docker and the bootJar/bootWar builds for me...

Comment (vorburger) [reply fineract-provider/build.gradle]: @vidakovic I've locally manually applied @ptuomola proposed 1 line change here (and my #1389; not directly related), and it seems to do the trick - I can `podman build .` it again with that (and had previously locally reproduced that it was broken before with the PR as is, just like on Travis). I hope you therefore don't mind if I Commit Suggestion here - I'm very curious to see if it will also pass on Travis! (I think it will.) If it does, you can still decide if you want to remove `$rootDir/`elsewhere, or just locally make this change (or pull & squash this) and force push it, so that we can merge it. (I won't merge it just yet, even if it passes with this change.)

Comment (vidakovic) [reply fineract-provider/build.gradle]: @vorburger Great, let's merge it then if there are no other issues left? Just verified the build again locally; works like a charm. Thanks again @ptuomola for sorting that one out!

**Review by vorburger — APPROVED** [review]
LGTM! 

@ptuomola OK for you as well - Approve? If yes, go ahead and merge - just remember to Squash on Merge (using GitHub UI) if you can, so the fix you proposed which I committed gets melded into Aleks original commit, instead of showing up separately on `develop`, that would be a tiny bit ugly.

**Comment by vidakovic** [comment]
@vorburger @ptuomola Don't want to be a pain... but could someone of you merge this please if you have a moment? Have some free cycles today and would like to proceed with FINERACT-1188. Thanks a ton.

**Comment by vorburger** [comment]
> @vorburger @ptuomola Don't want to be a pain... but could someone of you merge this please if you have a moment? 

I've LGTM it, but I think it's "fair" to give @ptuomola a chance to review it and LGTM is as well? Open Source can seem slow...

> Have some free cycles today and would like to proceed with FINERACT-1188

... but doesn't have to be! What I would do now in such a situation, if I were you, is to already start to `git rebase` your #1381 on this, and then the latest develop. That way, you can proceed at least locally. You could even also already push it to GitHub. GitHub is, unfortunately, a bit too stupid to understand that a PR is continuation of another one (FYI Gerrit does this MUCH better), but it still unblocks you, and reviewers can look at the N+ separate new commits after the "base" commits from another PR on a new PR;  and it means that once this PR is merged, even if there are still some small changes requested by another reviewer, your still ahead and will have an easier time to re-rebase other work that depends on this. Hope this approach makes sense?

**Comment by ptuomola** [comment]
@vidakovic I've been looking at this - the changes look OK to me, but when I import the project to Eclipse I get an "invalid external builder" error, indicating that something is broken about the OpenJPA enhance tool inclusion into Eclipse. Will take a look at this later tonight and let you know if I can work out what's wrong.

**Comment by vidakovic** [comment]
@ptuomola I can check quickly the issue with Eclipse... shouldn't be too complicated.

**Comment by vidakovic** [comment]
@ptuomola I would suggest to move on; the thing is that when all the changes are in place this will work nicely. I suggest that we move on to get the proper build configuration in place and not just this small step. Nevertheless, I'll still have a look at this. Would this be a blocker or can we merge anyway?

**Comment by ptuomola** [comment]
Well... I'm personally very interested in ensuring Eclipse continues to work, as I use it a lot! 

I've just looked at this and it's not immediately obvious to me how this can be solved:

The .externalToolBuilders directory should in my view be within fineract-provider, as it relates only to the fineract-provider subproject, not to any others. The root directory fineract is not a Eclipse-importable project, only fineract-provider is. 

But if I move it .externalToolBuilders back within fineract-provider, it still needs to refer to gradlew - which is now outside fineract-provider. And trying to do a relative path out of fineract-provider doesn't work.

So I'm not quite sure how this will work even with the rest of the changes incorporated. Maybe I'll take a look at the "full" branch next to see if that has the same problem, or if it has been resolved there...

**Comment by vidakovic** [comment]
@ptuomola Could you verify again? I think I found it... in my Eclipse no complaints about the launcher configuration.

**Comment by ptuomola** [comment]
I still get the same issue: 

Errors occurred during the build.
Errors running builder 'Integrated External Tool Builder' on project 'fineract-provider'.
Variable references non-existent resource : ${workspace_loc:/fineract-provider/gradlew}
Variable references non-existent resource : ${workspace_loc:/fineract-provider/gradlew}

So it's looking for gradlew inside fineract-provider, whereas it is now outside. Maybe we can add a symbolic link or something?

**Comment by ptuomola** [comment]
Ok - I've now found at least one possible way to fix this: 

- Add a symlink from fineract-provider/gradlew to ../gradlew
- Change the enhance task's classpath when running within eclipse to have fineract-provider as prefix

@vidakovic - maybe easiest if I push my changes into this pull request, and then we squash the commits? Assuming of course that I have the required permissions to do that. Or what would you prefer?

**Comment by vidakovic** [comment]
@ptuomola please feel free to push your changes

**Comment by ptuomola** [comment]
@vidakovic - done! This works at least for me in Eclipse - let's hope the build passes as well.

Now would be good to still squash the commits together. If I do it, I'm assuming it will then look like the whole commit was mine. Maybe you want to pull my changes and then squash the commits so that it's against your name?

**Comment by vidakovic** [comment]
@ptuomola done... your change is squashed. Thanks again for your help.

**Comment by vorburger** [comment]
> it will then look like the whole commit was mine. Maybe you want to pull my changes and then squash the commits so that it's against your name?

Just FYI for future cases like this (I'm NOT suggesting to change anything here anymore; @ptuomola feel free to merge when OK for you, as above), the traditional typical solution for that is a `Co-authored-by:` line - see e.g. https://docs.github.com/en/free-pro-team@latest/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors.

**Review by ptuomola — APPROVED** [review]
LGTM. Will merge once the build completes.

**Pull request merged**
